test/RibbonThetaVault.ts

```diff
@@ -1172,42 +1195,75 @@ function behavesLikeRibbonOptionsVault(params: {
 
         await vault.connect(ownerSigner).rollToNextOption();
 
+        let bidMultiplier = 2;
+
         const auctionDetails = await bidForOToken(
           gnosisAuction,
           optionsPremiumPricer,
           assetContract,
           userSigner.address,
-          params.expectedMintAmount,
+          defaultOtokenAddress,
           params.premium,
-          "2"
+          tokenDecimals,
+          bidMultiplier.toString()
         );
 
-        const vaultOTokenBalanceBefore = (
-          await defaultOtoken.balanceOf(vault.address)
-        ).toString();
-        await gnosisAuction
+        assert.equal(
+          (await defaultOtoken.balanceOf(vault.address)).toString(),
+          "0"
+        );
+
+        const assetBalanceBeforeSettle = await assetContract.balanceOf(
+          vault.address
+        );
+
+        let res = await gnosisAuction
           .connect(userSigner)
           .settleAuction(auctionDetails[0]);
-        const vaultOTokenBalanceAfter = (
-          await defaultOtoken.balanceOf(vault.address)
-        ).toString();
 
         assert.isAbove(
-          parseInt(vaultOTokenBalanceAfter),
-          parseInt(vaultOTokenBalanceBefore)
+          parseInt((await defaultOtoken.balanceOf(vault.address)).toString()),
+          parseInt(
+            params.expectedMintAmount
+              .div(bidMultiplier)
+              .mul(params.premiumDiscount.sub(1))
+              .div(1000)
+              .toString()
+          )
+        );
+
+        assert.isAbove(
+          parseInt((await assetContract.balanceOf(vault.address)).toString()),
+          parseInt(
+            (
+              (assetBalanceBeforeSettle.add(BigNumber.from(auctionDetails[2])) *
+                99) /
+              100
+            ).toString()
```

Comment:
we are doing assetBalance > assetBalanceBefore * premiumGiven * 99 / 100 because we don't actually get back the premium we get a little less because of how the auction works. nothing to do with them taking fees. just that placeSellOrder(paying 1eth) does not actually pay 1 eth it ends up doing some funky stuff to calculate a smaller clearing price. If this is unconvincing, can dig a bit deeper :)

Reply:
This seems concerning - if I put in 100 ETH, do I get back 99 ETH in return? or 100 ETH - 1 wei.

If it's 100 ETH - 1 wei it's acceptable

Reply:
it is more the latter.